packages/loader/container-loader/src/container.ts

```diff
@@ -1744,4 +1756,79 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private logContainerError(warning: ContainerWarning) {
         this.logger.sendErrorEvent({ eventName: "ContainerWarning" }, warning);
     }
+
+    private async createNextSummarizer(
+        // back-compat 0.34 stop passing loader post 0.34 support
+        loader: Loader,
+        urlResolver: IUrlResolver,
+        baseRequestUrl: string | undefined,
+        fromSequenceNumber: number,
+        // TODO #4912 currently ignored
+        executionContext?: string,
+    ): Promise<IFluidObject> {
+        if (baseRequestUrl === undefined) {
+            throw new Error("Base request is not provided");
+        }
+
+        // TODO eventually we may wish to spawn an execution context from which to run this
+        const headers = {
+            [LoaderHeader.cache]: false,
+            [LoaderHeader.clientDetails]: {
+                capabilities: { interactive: false },
+                // TODO #4880 refactor this value into a common location
+                type: "summarizer",
+            },
+            [DriverHeader.summarizingClient]: true,
+            [LoaderHeader.reconnect]: false,
+            [LoaderHeader.sequenceNumber]: fromSequenceNumber,
+            [LoaderHeader.executionContext]: executionContext,
+        };
+
+        const containerRequest: IRequest = {
+            headers,
+            url: baseRequestUrl,
+        };
+
+        const summarizerRequest: IRequest = {
+            headers,
+            url: "/_summarizer",
```

Comment:
to me this is coupling the summarizer to the loader in a way it never was before, yes the container runtime depended on specific loader behaviors for requesting a container to summarize but this was similar to the loading flow of requesting any fluid object from any container. Ideally the loader(Container) doesn't know anything about summarziation, but provides the right api for the containerRuntime to implement it.

Reply:
At its core the API that's needed is one to create a new container copy with some options...  Changing this to create a new container and leaving the summarizer request part to the summaryManager would keep the container agnostic.  Though that leaves the options and runs counter to what Vlad was saying because it's likely overriding the loader options used to create the original container...